api/apps/api/src/migrations/api/1628585506694-ScenarioSpecification.ts

```diff
@@ -0,0 +1,47 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+export class ScenarioSpecification1628585506694 implements MigrationInterface {
+  name = 'ScenarioSpecification1628585506694';
+
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(
+      `CREATE TABLE "specification" ("id" uuid NOT NULL, "scenario_id" uuid NOT NULL, "draft" boolean NOT NULL, CONSTRAINT "PK_01b2d90197e187e3187b2d888be" PRIMARY KEY ("id"))`,
+    );
+    await queryRunner.query(
+      `CREATE TABLE "specification_feature" ("id" uuid NOT NULL DEFAULT uuid_generate_v4(), "specification_feature_config_id" uuid NOT NULL, "feature_id" uuid NOT NULL, "calculated" boolean NOT NULL, CONSTRAINT "PK_fe6096ea6f25997b62b8a5c5f99" PRIMARY KEY ("id"))`,
+    );
+    await queryRunner.query(
+      `CREATE TYPE "specification_feature_config_operation_enum" AS ENUM('split', 'stratification', 'copy')`,
+    );
+    await queryRunner.query(
+      `CREATE TABLE "specification_feature_config" ("id" uuid NOT NULL DEFAULT uuid_generate_v4(), "specification_id" uuid NOT NULL, "base_feature_id" uuid NOT NULL, "against_feature_id" uuid, "operation" "specification_feature_config_operation_enum" NOT NULL, "features_determined" boolean NOT NULL, CONSTRAINT "PK_df1e25cf3972f83b1ff5bcb0337" PRIMARY KEY ("id"))`,
+    );
+    await queryRunner.query(
+      `ALTER TABLE "specification" ADD CONSTRAINT "FK_2cecab1b81e5fa64f2fdbbaec76" FOREIGN KEY ("scenario_id") REFERENCES "scenarios"("id") ON DELETE CASCADE ON UPDATE NO ACTION`,
```

Comment:
should we cascade on update too? probably moot in practice as we don't support updating `scenarios.id`, but i'd say it doesn't hurt to have the cascade (same for the other FKs below)

Reply:
TBH we should rather avoid doing such operations directly on the entity if we want to keep the system consistent.

Reply:
yes, we won't be updating UUID ids in general (if that's what you mean here). In general I see no reason for *not* having cascades on both delete and update unless there is a specific case for keeping FKs constant despite changes (for whatever reason) in the referenced IDs or a need for micro optimizations in terms of performance in specific contexts/use cases

Reply:
but yes, since we do have a peculiar setup with non-really-FKs across dbs, I agree that in this case restricting the update may be preferable, so ok.